lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -125,14 +156,85 @@ def close
     # * `:path` - String. The leading path for prefixing Elasticsearch
     #   requests. This is sometimes used if you are proxying Elasticsearch access
     #   through a special http path, such as using mod_rewrite.
-    def build_pool(options)
-      hosts = options[:hosts] || ["127.0.0.1"]
-      client_settings = options[:client_settings] || {}
-      timeout = options[:timeout] || 0
+    def calculate_property(uris, property, default, sniff_check)
+      values = uris.map(&property).uniq
+
+      if sniff_check && values.size > 1
+          raise LogStash::ConfigurationError, "Cannot have multiple values for #{property} in hosts when sniffing is enabled!"
+      end
+
+      uri_value = values.first
+
+      default = nil if default == '' # Blanks are as good as nil
+      uri_value = nil if uri_value == ''
+
+      if default && uri_value && (default != uri_value)
```

Comment:
Earlier we check that `values` has a length of one, you can't mix and match the value described here. Then we check that the first (and by now we know only) value is also the same as the value in the URI.